app/javascript/Greetings.js

```diff
@@ -0,0 +1,23 @@
+import React, { useState, useEffect } from "react";
+
+const Greetings = () => {
+  const [randomGreeting, setRandomGreeting] = useState("");
+
+  useEffect(() => {
+    fetch("/greetings.json")
+      .then((response) => response.json())
+      .then((data) => {
+        setRandomGreeting(data[0]?.body || "");
+      })
+      .catch((error) => console.error("Error fetching greetings:", error));
+  }, []);
```

Comment:
- [ ] Let be break this down for you to uderstand what the other reviwers meant. 
- Based on the project requirements, You're to use  redux for data state management: The project requirements specifically mention the creation of a `store`, an `action`, and a reducer to connect with your API endpoint for retrieving random messages. 
- The above means, the fetching of data should be from the API endpoint which your rails application will run on(same port). This is to show the connectio between the front and back end 

### Sample code:
```javascript
import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import axios from 'axios';

const GREETINGS_URL = 'http://127.0.0.1:3000/api/v1/greeting';

export const getGreetings = createAsyncThunk(
  'greetings/getGreetings',
  async (_, thunkAPI) => {
    try {
      const resp = await axios.get(GREETINGS_URL);
      return resp.data;
    } catch (error) {
      return thunkAPI.rejectWithValue(error);
    }
  },
);
```